### PR TITLE
added semver official regex pattern

### LINKED ISSUE #4453: [Bug] dbt_project.yml does not fully support semantic versioning
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

One can read [in the docs](https://docs.getdbt.com/reference/project-configs/version) that semantic versioning is supported. If so, pre-versions like 1.0.0-pre.1 should be supportes (as the link in the docs says).
Currently the version number causes a Runtime Error:
```
Encountered an error while reading the project:
  ERROR: Runtime Error
  at path ['version']: '0.0.0-pre.1' is not valid under any of the given schemas
```

```
# dbt_project.yml

version: '1.0.0-pre.1'
```

### Expected Behavior

Version 1.0.0-pre.1 should be an accepted version, because it follows the rules of [semantic versioning](https://semver.org/#backusnaur-form-grammar-for-valid-semver-versions)

### Steps To Reproduce

_No response_

### Relevant log output

_No response_

### Environment

```markdown
- OS: macOS 11.5.2
- Python: 3.9.7
- dbt: 21.1
```

### What database are you using dbt with?

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by alswang18**
resolves #4453

### Description

Added support for full semver without breaking support for 2 group versioning

### Checklist

- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] I have updated the `CHANGELOG.md` and added information about my change

### CODE REVIEW

**Comment by b0lle** [comment]
I tested some cases on regex101.com. Looks legit to me

Unfortunately there seems to be no opportunity to extend some test cases :/

**Comment by b0lle** [comment]
Great. That should do the trick :)

**Comment by alswang18** [comment]
@b0lle Should be better now
image

**Comment by alswang18** [comment]
Also, I signed the CLA over the weekend, wondering why the bot wont recognize me.

**Comment by leahwicz** [comment]
@cla-bot check

**Comment by alswang18** [comment]
> @alswang18 it looks like your formatting was a bit off when you signed the CLA. It should be just your username alswang18, no @ or github url. If you sign with that format you should be good to go!

Just tried it again

**Comment by alswang18** [comment]
@cla-bot check

**Review by gshank — APPROVED** [review]
Looks good.

**Comment by alswang18** [comment]
@gshank Can we merge this in?

**Comment by leahwicz** [comment]
@alswang18 could you do one last thing and add an entry in the [Changelog](https://github.com/dbt-labs/dbt-core/blob/main/CHANGELOG.md) file for this change. It would be under section `1.1.0`. Also add yourself as a contributor!

**Comment by alswang18** [comment]
Got it, just added @leahwicz

File: CHANGELOG.md [hunk 1]
```diff
@@ -15,6 +16,8 @@
 
 Contributors:
 - [@NiallRees](https://github.com/NiallRees) ([#4447](https://github.com/dbt-labs/dbt-core/pull/4447))
+- [@alswang18](https://github.com/alswang18) ([#4644]
+https://github.com/dbt-labs/dbt-core/pull/4644))
```
Comment (leahwicz) [inline CHANGELOG.md:20]:
```suggestion
- [@alswang18](https://github.com/alswang18) ([#4644](https://github.com/dbt-labs/dbt-core/pull/4644))
```

Comment (leahwicz) [reply CHANGELOG.md:20]: Just fixing the broken link and then this LGTM!

Comment (alswang18) [reply CHANGELOG.md]: Added the change :)

**Comment by leahwicz** [comment]
Thanks @alswang18 for the contribution!!!

**Comment by leahwicz** [comment]
Some GitHub Checks funkiness so closing and reopening to kick them off

**Comment by alswang18** [comment]
@leahwicz sure, when do we merge?

**Pull request merged**